include/zephyr/devicetree.h

```diff
@@ -3755,6 +3755,40 @@
 #define DT_INST_STRING_UNQUOTED_OR(inst, name, default_value) \
 	DT_STRING_UNQUOTED_OR(DT_DRV_INST(inst), name, default_value)
 
+/**
+ * @brief Test if any given compat is on a bus of a given type
+ *        and has status okay
+ *
+ * In usually, the DT_ANY_INST_ON_BUS_STATUS_OKAY can use for check for node
+ * that is on specified bus existence check.
+ * This is useful for handle multiple compat in single source file.
+ *
+ * Example devicetree overlay:
+ *
+ * @code{.dts}
+ *     &i2c0 {
+ *            temp: temperature-sensor@76 {
+ *                     compatible = "vnd,some-sensor";
+ *                     reg = <0x76>;
+ *            };
+ *     };
+ * @endcode
+ *
+ * Example usage, assuming `i2c0` is an I2C bus controller node, and
+ * therefore `temp` is on an I2C bus:
+ *
+ * @code{.c}
+ *     DT_COMPAT_ANY_ON_BUS(vnd_some_sensor, i2c) // 1
+ * @endcode
+ *
+ * @param compat lowercase-and-underscores compatible, without quotes
+ * @param bus a binding's bus type as a C token, lowercased and without quotes
+ * @return 1 if any enabled node with that compatible is on that bus type,
+ *         0 otherwise
+ */
+#define DT_COMPAT_ANY_ON_BUS(compat, bus) \
```

Comment:
Please rename this to DT_HAS_COMPAT_ON_BUS_STATUS_OKAY for consistency with other APIs like DT_HAS_COMPAT_STATUS_OKAY.

Reply:
I renamed it to DT_*ANY*_HAS_COMPAT_ON_BUS_STATUS_OKAY, considering symmetry with DT_ANY_INST_ON_BUS_STATUS_OKAY.
The only difference is that the argument is instance number or compatibility, so I changed it accordingly.

Reply:
I think the symmetry with `DT_ANY_INST` is not clear. "ANY_INST" means "any instance of DT_DRV_COMPAT". "ANY_HAS" does not make sense to me.

Reply:
I reconsidered, and I got it you make sense.
I applied it, Thank you.